docs-source/content/userguide/managing-domains/domain-lifecycle/introspection.md

```diff
@@ -0,0 +1,70 @@
+domain can be updated without restarting any servers. Currently, this is not supported for 
+the Model in Image [domain home source type]({{< relref "/userguide/managing-domains/choosing-a-model/_index.md" >}}).
+
+#### Common re-introspection scenarios
+
+This document describes what actions allow avoiding the restart of servers in a number of common re-introspection scenarios:
+
+* Modifying the WebLogic configuration
+* Changing the custom domain configuration overrides (also called situational configuration) for Domain in PV and Domain in Image domains
+
+### Use cases
+
+#### Modifying the WebLogic Server configuration
+
+Changes to the Oracle WebLogic Server configuration may not require a full restart, depending on the domain home location 
+and the type of configuration change.
+
+* **Domain in PV:**
+For a domain home on PV, it is often possible to avoid a full restart when making a dynamic change.
+    * Configuration changes that will not restart any servers:
+      * Adding a cluster, server or dynamic server
+      * Changing a dynamic property such as a server's connection timeout property
+    * Configuration changes that may shut down or restart some servers:  
+      * Removing a cluster, server, dynamic server, or network access point
+      * Changing a cluster, server, dynamic server, or network access point name
+      * Enabling or disabling the listen port, SSL port, or admin port
+      * Changing any port numbers
+      * Changing a network access point's public address
+
+#### Changing the custom domain configuration overrides
+
+Changes to configuration overrides may be adopted without restarting the domain. These changes are made by:
+  * Changing the domain resource's `configuration.overridesConfigMap` to point to a different configuration map
+  * Changing the domain resource's `configuration.secrets` to point to a different list of secrets
+  * Changing the contents of the configuration map referenced by `configuration.overridesConfigMap`
+  * Changing the contents to any of the secrets referenced by `configuration.secrets`
+  
+The running servers will, however, only see the actual changes when restarted.  
+<!-- Note: this will change when the "Implement distribution strategy for config override changes" feature is merged. --> 
+
+### Forcing the operator to rerun the introspector job
+
+The operator will run the introspector job when the domain specification field `introspectVersion` is changed. To
+edit this field in the domain resource directly using the `kubectl` command-line tool:
+
+```
+kubectl edit domain <domain name> -n <domain namespace>
+```
+
+The `edit` command opens a text editor which lets you edit the domain resource in place.
+
+{{% notice note %}}
+Typically, it's better to edit the domain resource directly; otherwise, if you scaled the domain, and you edit only the original `domain.yaml` file and reapply it, you could go back to your old replicas count.
+{{% /notice %}}
```

Comment:
(1) I don't understand the note.  If you modify something like a replicas count, it's usually better to edit the original YAML file so that the changed replica count isn't lost - but the note seems to say the opposite.   
(2) It'd be helpful here to show the kubectl commands for getting the current introspectVersion and patching it. kubectl edit isn't suitable for automation...   See the 'restartVersion' doc in the MII `runtime-updates.md`.

```
#### Changing a domain resource `restartVersion`

As was mentioned in the [overview](#overview), one way to tell the operator to apply your configuration changes to a running domain is by altering the domain resource `restartVersion`. Here are some common ways to do this:

 - You can alter `restartVersion` interactively using `kubectl edit -n MY_NAMESPACE domain MY_DOMAINUID`.

 - If you have your domain's resource file, then you can alter this file and call `kubectl apply -f` on the file.

 - You can use the Kubernetes `get` and `patch` commands. Here's a sample automation script that takes a namespace as the first parameter (default `sample-domain1-ns`) and that takes a domainUID as the second parameter (default `sample-domain1`):

   ```
   #!/bin/bash
   NAMESPACE=${1:-sample-domain1-ns}
   DOMAINUID=${2:-sample-domain1}
   currentRV=$(kubectl -n ${NAMESPACE} get domain ${DOMAINUID} -o=jsonpath='{.spec.restartVersion}')
   if [ $? = 0 ]; then
     # we enter here only if the previous command succeeded

     nextRV=$((currentRV + 1))

     echo "@@ Info: Rolling domain '${DOMAINUID}' in namespace '${NAMESPACE}' from restartVersion='${currentRV}' to restartVersion='${nextRV}'."

     kubectl -n ${NAMESPACE} patch domain ${DOMAINUID} --type='json' \
       -p='[{"op": "replace", "path": "/spec/restartVersion", "value": "'${nextRV}'" }]'
   fi
   ```

Reply:
It takes a lot of text to describe how to use `kubectl edit`. I appreciate the note about an original source of truth, but if we're going to give the command for a change to the live document then we might as well give the `kubectl patch` command.